Sources/GRPCCore/Call/Server/Internal/ServerRPCExecutor.swift

```diff
@@ -0,0 +1,299 @@
+          ()
+
+        case .executed:
+          // The work finished. Cancel any remaining tasks.
+          group.cancelAll()
+        }
+      }
+    }
+  }
+
+  @inlinable
+  static func _processRPC<Input, Output>(
+    method: MethodDescriptor,
+    metadata: Metadata,
+    inbound: UnsafeTransfer<RPCAsyncSequence<RPCRequestPart>.AsyncIterator>,
+    outbound: RPCWriter<RPCResponsePart>.Closable,
+    deserializer: some MessageDeserializer<Input>,
+    serializer: some MessageSerializer<Output>,
+    interceptors: [any ServerInterceptor],
+    handler: @escaping @Sendable (
+      ServerRequest.Stream<Input>
+    ) async throws -> ServerResponse.Stream<Output>
+  ) async {
+    let messages = AsyncIteratorSequence(inbound.wrappedValue).map { part throws -> Input in
+      switch part {
+      case .message(let bytes):
+        return try deserializer.deserialize(bytes)
+      case .metadata:
+        throw RPCError(
+          code: .internalError,
+          message: """
+            Server received an extra set of metadata. Only one set of metadata may be received \
+            at the start of the RPC. This is likely to be caused by a misbehaving client.
+            """
+        )
+      }
+    }
+
+    let response = await Result {
+      // Run the request through the interceptors, finally passing it to the handler.
+      return try await Self._intercept(
+        request: ServerRequest.Stream(
+          metadata: metadata,
+          messages: RPCAsyncSequence(wrapping: messages)
+        ),
+        context: ServerInterceptorContext(descriptor: method),
+        interceptors: interceptors
+      ) { request, _ in
+        try await handler(request)
+      }
+    }.castError(to: RPCError.self) { error in
+      RPCError(code: .unknown, message: "Service method threw an unknown error.", cause: error)
+    }.flatMap { response in
+      response.accepted
+    }
+
+    let status: Status
+    let metadata: Metadata
+
+    switch response {
+    case .success(let contents):
+      let result = await Result {
+        // Write the metadata and run the producer.
+        try await outbound.write(.metadata(contents.metadata))
+        return try await contents.producer(
+          .serializingToRPCResponsePart(into: outbound, with: serializer)
+        )
+      }.castError(to: RPCError.self) { error in
+        RPCError(code: .unknown, message: "", cause: error)
+      }
+
+      switch result {
+      case .success(let trailingMetadata):
+        status = .ok
+        metadata = trailingMetadata
+      case .failure(let error):
+        status = Status(code: Status.Code(error.code), message: error.message)
+        metadata = error.metadata
+      }
+
+    case .failure(let error):
+      status = Status(code: Status.Code(error.code), message: error.message)
+      metadata = error.metadata
+    }
+
+    try? await outbound.write(.status(status, metadata))
```

Comment:
At this point the RPC handle has completed, interceptors are done etc., so the only interested party at this point is the client. The only reason this write should fail is if the stream is closed (likely because the client or server closed or is closing). So there's nothing we can do about this and I don't think logging would even be all that helpful here.